core/server/master/src/main/java/alluxio/underfs/UfsStatusCache.java

```diff
@@ -120,9 +123,12 @@ public Collection<UfsStatus> addChildren(AlluxioURI path, Collection<UfsStatus>
       AlluxioURI childPath = path.joinUnsafe(child.getName());
       addStatus(childPath, child);
     });
-    // Update global counters for all InodeSyncStream
-    DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_CHILDREN_SIZE_TOTAL.inc(children.size());
-    return mChildren.put(path, children);
+    Collection<UfsStatus> previousStatuses = mChildren.put(path, children);
+    if (previousStatuses == null) {
+      // Update global counters for all InodeSyncStream
+      DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_CHILDREN_SIZE_TOTAL.inc(children.size());
+    }
+    return previousStatuses;
```

Comment:
if return is not null, the new children collection will override the old children collection, i wonder if the size will stay unchanged.

Reply:
Good catch.

Reply:
addresses in [8ad7c9b](https://github.com/Alluxio/alluxio/pull/15370/commits/8ad7c9baaa5062de73370413dcc1c7835f5ecab0)

Reply:
`inc()` and `dec()` do accept negative numbers, so this is okay